### ISSUE TITLE
Add \s to lark.lexer._regexp_has_newline

### ISSUE THREAD

**Issue by fabiommendes** [lark-parser/lark#420]
This is a trivial fix: _regexp_has_newline does not test for \s patterns in the regex, so if you use it in your grammar, Lark will report wrong line numbers. Should I send a PR?

**Comment by erezsh** [comment]
Sure, sounds good

**Comment by erezsh** [comment]
Make sure to include a test, proving this issue

**Comment by fabiommendes** [comment]
I made [3 commits](https://github.com/fabiommendes/lark/commits/master) (1st proving the issue, 2nd correcting it and 3rd adding .idea to gitignore) and I noticed that my editor automatically reformatted the code for better pep8 compliance. Should Lark adopt [black](https://github.com/psf/black) or something similar to avoid those issues?

I can re-create the commits sans reformatting, but I think it would be better to have a well defined style guide for Lark. I use [pre-commit](https://github.com/pre-commit/pre-commit) with Black a lot in my workflow and it is a great way to ensure style consistency and avoid trivial regressions and errors. I can configure Travis to check that that if you want.

**Comment by erezsh** [comment]
I don't like reformatters. Each one has its own concept of what is "right" and what isn't, and it enforces it unintelligently, without taking into account context and circumstances. Personally, I feel that humans are more than capable of maintaining a coherent code-base, even if not everything is exactly styled through a collection of simple rules. Not sure what sort of trivial regressions you're thinking of.

Regarding Lark's code style, it's basically pep8, except when it doesn't make any sense. If there's anything *specific* about Lark's style that you think could be better, you're more than welcome to make that argument.

The rest of the commits are fine, I'll accept them in a pull-request.

**Comment by fabiommendes** [comment]
I recreated the commits without reformatting and put the test on the factory test class that checks all configurations of parsers and lexers. I think now it is good to go!

As for code formatters, I personally think they are a great relief, and they tend to make me more productive. I changed my mind after using Elm for a while. It has a builtin formatter that accepts absolutely no configuration option and uses very strange formatting rules when compared to Python. It changed from disgust, to acceptance to really loving it!  But each to each own :)
